cli/azd/pkg/tools/github/github.go

```diff
@@ -84,25 +101,43 @@ func (cli *ghCli) InstallUrl() string {
 	return "https://aka.ms/azure-dev/github-cli-install"
 }
 
-func (cli *ghCli) CheckAuth(ctx context.Context, hostname string) (bool, error) {
-	runArgs := exec.NewRunArgs("gh", "auth", "status", "--hostname", hostname)
+// The result from calling CheckAuth
+type CheckAuthResult struct {
+	LoggedIn    bool
+	TokenSource AuthTokenSource
+}
+
+// The source of the auth token used by `gh` CLI
+type AuthTokenSource int
+
+const (
+	TokenSourceFile AuthTokenSource = iota
+	// See isEnvVarTokenSourceRegex for example token env vars
+	TokenSourceEnvVar
+)
+
+func (cli *ghCli) CheckAuth(ctx context.Context, hostname string) (CheckAuthResult, error) {
```

Comment:
Should it be `GetAuthStatus` or something like that?   The verb `Check` is usually for returning just a `bool` value, similar to `Is` or `Has`

Reply:
Makes sense